src/routes/project/components/TextEditor.js

```diff
@@ -0,0 +1,194 @@
+import React, { Component } from 'react';
+import Editor from 'draft-js-plugins-editor';
+import { EditorState, RichUtils, getVisibleSelectionRect, convertToRaw, convertFromRaw } from 'draft-js'
+import createCounterPlugin from 'draft-js-counter-plugin';
+
+const counterPlugin = createCounterPlugin();
+const { CharCounter, WordCounter, CustomCounter } = counterPlugin;
+const plugins = [counterPlugin];
+
+
+class StyleButton extends Component {
+  constructor() {
+    super();
+    this.onToggle = (e) => {
+      e.preventDefault();
+      this.props.onToggle(this.props.style);
+    };
+  }
+
+  render() {
+    let className = 'toolbarItem';
+    if (this.props.active) {
+      className += ' active';
+    }
+    return (
+      <span className={className} onMouseDown={this.onToggle}>
+        {this.props.label}
+      </span>
+    );
+  }
+}
+
+const styleTypes = [
+  {label: 'Bold', style: 'BOLD'},
+  {label: 'Italic', style: 'ITALIC'},
+  {label: 'Underline', style: 'UNDERLINE'},
+  {label: 'Strikethrough', style: 'STRIKETHROUGH'}
+];
+
+const condensedStyleTypes = [
+  {label: 'B', style: 'BOLD'},
+  {label: 'I', style: 'ITALIC'},
+  {label: 'U', style: 'UNDERLINE'},
+  {label: 'S', style: 'STRIKETHROUGH'}
+];
+
+const InlineStyleControls = (props) => {
+  const currentStyle = props.editorState.getCurrentInlineStyle();
+
+  return (
+    <div className="InlineStyleControls">
+      {!props.condensed ?
+        styleTypes.map(type =>
+        <StyleButton
+          key={type.label}
+          active={currentStyle.has(type.style)}
+          label={type.label}
+          onToggle={props.onToggle}
+          style={type.style}
+        />
+      )
+      :
+      condensedStyleTypes.map(type =>
+      <StyleButton
+        key={type.label}
+        active={currentStyle.has(type.style)}
+        label={type.label}
+        onToggle={props.onToggle}
+        style={type.style}
+      />
+    )}
+    </div>
```

Comment:
you're right, added an `initial' field to the styleTypes object and did the conditional on that prop only:

```const styleTypes = [
  {initial: 'B', label: 'Bold', style: 'BOLD'},
  {initial: 'I', label: 'Italic', style: 'ITALIC'},
  {initial: 'U', label: 'Underline', style: 'UNDERLINE'},
  {initial: 'S', label: 'Strikethrough', style: 'STRIKETHROUGH'}
];

const InlineStyleControls = (props) => {
  const currentStyle = props.editorState.getCurrentInlineStyle();
  const { condensed } = props;

  return (
    <div className="InlineStyleControls">
      {styleTypes.map(type =>
        <StyleButton
          key={type.label}
          active={currentStyle.has(type.style)}
          label={condensed ? type.initial : type.label}
          onToggle={props.onToggle}
          style={type.style}
        />
      )}
    </div>
  );
};```